File: app/src/main/java/com/example/proyecto_final/TareasActivity.java
package com.example.proyecto_final;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.firestore.DocumentChange;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.EventListener;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreException;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;

public class TareasActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    ArrayList<Tarea> tareasArrayList;
    TareasAdapter tareasAdapter;

    ProgressDialog progressDialog;
    ImageButton btnCrear, btnAtras;

    FirebaseAuth auth = FirebaseAuth.getInstance();

    // Obtiene el usuario actualmente conectado
    FirebaseUser currentUser = auth.getCurrentUser();



    FirebaseFirestore db = FirebaseFirestore.getInstance();


    private DatabaseReference mDatabase;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tareas);

        progressDialog = new ProgressDialog(this);
        progressDialog.setCancelable(false);
        progressDialog.setMessage("Obteniendo los datos...");
        progressDialog.show();

        recyclerView = findViewById(R.id.my_recyclerview);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        tareasArrayList = new ArrayList<Tarea>();
        tareasAdapter = new TareasAdapter(TareasActivity.this,tareasArrayList);

        recyclerView.setAdapter(tareasAdapter);

        btnCrear = findViewById(R.id.crearButton);


        mDatabase = FirebaseDatabase.getInstance().getReference();


        // Si el usuario está conectado, muestra su información
        if (currentUser != null) {
            String email = currentUser.getEmail();


            // Actualiza la interfaz de usuario con la información del usuario
            // ...


            db.collection("Perfiles")
                    .get()
                    .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
                        @Override
                        public void onComplete(@NonNull Task<QuerySnapshot> task) {
                            if (task.isSuccessful()) {
                                for (QueryDocumentSnapshot document : task.getResult()) {
                                    Log.d("Seccessful:", document.getId() + " => " + document.getData());

                                    if (email.equals(document.getData().get("email-perfil"))) {
                                        TextView textView = findViewById(R.id.user_name);

                                        textView.setText((CharSequence) document.getData().get("nombre_perfil"));
                                    }
                                }
                            } else {
                                Log.w("Error getting documents.", task.getException());
                            }
                        }
                    });
        }











        btnCrear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Realiza la navegación a la otra pantalla
                Intent intent = new Intent(TareasActivity.this, CrearTareaActivity.class);
                startActivity(intent);
            }
        });

        btnAtras = findViewById(R.id.btn_atras);

        btnAtras.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Realiza la navegación a la otra pantalla
                Intent intent = new Intent(TareasActivity.this, HomeActivity.class);
                startActivity(intent);
            }
        });

        EventChangeListener();

    }

    private void EventChangeListener() {

        db.collection("Tareas")
                .addSnapshotListener(new EventListener<QuerySnapshot>() {
                    @Override
                    public void onEvent(@Nullable QuerySnapshot value, @Nullable FirebaseFirestoreException error) {
                        if (error != null){

                            if (progressDialog.isShowing())
                                progressDialog.dismiss();
                            Log.e("Error al conectar la Base de Datos...",error.getMessage());
                            return;
                        }
                        for (DocumentChange dc : value.getDocumentChanges()){

                            if (dc.getType() == DocumentChange.Type.ADDED){

                                tareasArrayList.add(dc.getDocument().toObject(Tarea.class));

                            }

                            tareasAdapter.notifyDataSetChanged();
                            if (progressDialog.isShowing())
                                progressDialog.dismiss();

                            }
                        }
                });
        }

}

File: app/src/main/java/com/example/proyecto_final/TareasActivity.kt
package com.example.proyecto_final

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.firestore.DocumentChange
import com.google.firebase.firestore.EventListener
import com.google.firebase.firestore.FirebaseFirestore

class TareasActivity : AppCompatActivity() {
    var recyclerView: RecyclerView? = null
    var tareasArrayList: ArrayList<Tarea>? = null
    var tareasAdapter: TareasAdapter? = null
    var progressDialog: ProgressDialog? = null
    var btnCrear: ImageButton? = null
    var btnAtras: ImageButton? = null
    var auth = FirebaseAuth.getInstance()

    // Obtiene el usuario actualmente conectado
    var currentUser = auth.currentUser
    var db = FirebaseFirestore.getInstance()
    private var mDatabase: DatabaseReference? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tareas)
        progressDialog = ProgressDialog(this)
        progressDialog!!.setCancelable(false)
        progressDialog!!.setMessage("Obteniendo los datos...")
        progressDialog!!.show()
        recyclerView = findViewById(R.id.my_recyclerview)
        recyclerView.setHasFixedSize(true)
        recyclerView.setLayoutManager(LinearLayoutManager(this))
        tareasArrayList = ArrayList()
        tareasAdapter = TareasAdapter(this@TareasActivity, tareasArrayList)
        recyclerView.setAdapter(tareasAdapter)
        btnCrear = findViewById(R.id.crearButton)
        mDatabase = FirebaseDatabase.getInstance().reference


        // Si el usuario está conectado, muestra su información
        if (currentUser != null) {
            val email = currentUser!!.email


            // Actualiza la interfaz de usuario con la información del usuario
            // ...
            db.collection("Perfiles")
                .get()
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        for (document in task.result) {
                            Log.d("Seccessful:", document.id + " => " + document.data)
                            if (email == document.data["email-perfil"]) {
                                val textView = findViewById<TextView>(R.id.user_name)
                                textView.text = document.data["nombre_perfil"] as CharSequence?
                            }
                        }
                    } else {
                        Log.w("Error getting documents.", task.exception)
                    }
                }
        }
        btnCrear.setOnClickListener(View.OnClickListener { // Realiza la navegación a la otra pantalla
            val intent = Intent(this@TareasActivity, CrearTareaActivity::class.java)
            startActivity(intent)
        })
        btnAtras = findViewById(R.id.btn_atras)
        btnAtras.setOnClickListener(View.OnClickListener { // Realiza la navegación a la otra pantalla
            val intent = Intent(this@TareasActivity, HomeActivity::class.java)
            startActivity(intent)
        })
        EventChangeListener()
    }

    private fun EventChangeListener() {
        db.collection("Tareas")
            .addSnapshotListener(EventListener { value, error ->
                if (error != null) {
                    if (progressDialog!!.isShowing) progressDialog!!.dismiss()
                    Log.e("Error al conectar la Base de Datos...", error.message!!)
                    return@EventListener
                }
                for (dc in value!!.documentChanges) {
                    if (dc.type == DocumentChange.Type.ADDED) {
                        tareasArrayList!!.add(dc.document.toObject(Tarea::class.java))
                    }
                    tareasAdapter!!.notifyDataSetChanged()
                    if (progressDialog!!.isShowing) progressDialog!!.dismiss()
                }
            })
    }
}